components/toast_wrapper/toast_wrapper.tsx

```diff
@@ -2,103 +2,106 @@
-         * Object from react-router
-         */
-        match: PropTypes.shape({
-            params: PropTypes.shape({
-                team: PropTypes.string,
-            }).isRequired,
-        }).isRequired,
-
-        actions: PropTypes.shape({
-
-            /**
-             * Action creator to update toast status
-             */
-            updateToastStatus: PropTypes.func.isRequired,
-        }).isRequired,
+export type Props = WrappedComponentProps & RouteComponentProps<{team: string}> & {
+    channelMarkedAsUnread?: boolean;
+    postListIds: string[];
+    latestPostTimeStamp?: number;
+    atBottom: boolean | null;
+    lastViewedBottom: number;
+    width: number;
+    focusedPostId?: string;
+    initScrollOffsetFromBottom: number;
+    updateNewMessagesAtInChannel: (lastViewedAt?: number) => void;
+    scrollToNewMessage: () => void;
+    scrollToLatestMessages: () => void;
+    scrollToUnreadMessages: () => void;
+    updateLastViewedBottomAt: (lastViewedBottom?: number) => void;
+    showSearchHintToast: boolean;
+    onSearchHintDismiss: () => void;
+    shouldStartFromBottomWhenUnread: boolean;
+    isNewMessageLineReached: boolean;
+    rootPosts: Record<string, boolean>;
+    lastViewedAt: number;
+    newRecentMessagesCount: number;
+    unreadScrollPosition: string;
+    isCollapsedThreadsEnabled: boolean;
+    unreadCountInChannel: number;
+    atLatestPost?: boolean;
+    channelId: string;
+    intl: IntlShape;
+    actions: {
+        updateToastStatus: (status: boolean) => void;
     };
-
+};
+
+type State = {
+    unreadCount: number;
+    unreadCountInChannel: number;
+    channelMarkedAsUnread?: boolean;
+    lastViewedAt?: number;
+    showUnreadToast?: boolean;
+    showNewMessagesToast?: boolean;
+    showMessageHistoryToast?: boolean;
+    showUnreadWithBottomStartToast?: boolean;
+};
+
+export class ToastWrapperClass extends React.PureComponent<Props, State> {
+    mounted?: boolean;
     static defaultProps = {
         focusedPostId: '',
     };
 
-    constructor(props) {
+    constructor(props: Props) {
         super(props);
         this.state = {
             unreadCountInChannel: props.unreadCountInChannel,
+            unreadCount: 0,
         };
     }
 
-    static countNewMessages = (postListIds, rootPosts, isCollapsedThreadsEnabled) => {
+    static countNewMessages = (rootPosts: Record<string, boolean>, isCollapsedThreadsEnabled: boolean, postListIds: string[] = []) => {
         const mark = getNewMessageIndex(postListIds);
         if (mark <= 0) {
             return 0;
         }
-        let newMessages = postListIds.slice(0, mark).filter((id) => !isIdNotPost(id));
+        let newMessages = postListIds?.slice(0, mark).filter((id) => !isIdNotPost(id));
         if (isCollapsedThreadsEnabled) { // in collapsed mode we only count root posts
-            newMessages = newMessages.filter((id) => rootPosts[id]);
+            newMessages = newMessages?.filter((id) => rootPosts[id]);
```

Comment:
Same as above, since `newMessages` is an array (this also applies to the return statement below).